### PR TITLE
Chore/remove rails from required libraries

### LINKED ISSUE #11: Is Rails dependency really necessary?
First of all, thank you for this gem. :)

But is it really necessary to pull Rails as a dependency? I see that it's specified as "development" dependency in `gemspec` file, but there's an explicit `require 'rails'` statement in `lib/grape_fast_jsonapi.rb:3`, which prevents it from being used with, for example, Sinatra-based apps.

### PR DESCRIPTION
**Pull request by EmCousin**
This PR will close #11 

- [x] Removing an unnecessary call to `rails`. We're keeping Rails as a development dependency at the moment
- [x] Fixed potential security issue with Nokogiri
- [x] Provides a more intuitive ability to use a custom serializer when rendering an object

### CODE REVIEW

**Comment by spijet** [comment]
I see a weird error when sending a hash with an empty array inside as a response — the app catches a `NoMethodError` in the Formatter class. Is it expected?
Please check [this gist](https://gist.github.com/spijet/7aa95e3351394b79ec8e179df2b61e96) for minimal app code and log.

**Comment by EmCousin** [comment]
@spijet Interesting. After digging a bit it appears that a forked version fixed that: https://github.com/cartedo/grape_fast_jsonapi/commit/79049c1934dbd5cb1e3e8b074cc1a171b9f9f766 

I'll see for updating the PR with adapted testing cover

**Comment by spijet** [comment]
Yep, this seems to fix the problem with empty arrays.

I also ran into a problem with `fast_jsonapi_serializable()` method, but it's caused by my own stuff — the app relies on module namespacing, so all serializer classes are named as `AppName::Serializers::<Model>` instead of `<Model>Serializer`. I'll probably just rewrite the method to make it work with this scheme and bundle it as an extended class. :)

**Comment by spijet** [comment]
Actually, I was wrong, and my app *does* use `<Model>Serializer` classes, but it still holds them in `AppName::Serializers::` namespace. So, here's what worked for me in the end:
```ruby
def serializable_class(object)
  object = object.first if object.is_a?(Array)
  klass = object.model_name.name.split('::').last
  format('%<klass>sSerializer', klass: klass).safe_constantize
end
```

It probably won't work for everyone, as it (most likely) relies on auto-loading and auto-resolving magic made possible by Zeitwerk gem, but at least it won't fail on non-namespaced classes.

**Comment by spijet** [comment]
A little addition to my take on `serializable_class()` — in case ActiveSupport is available, we can use [ActiveSupport::Inflector.demodulize](https://www.rubydoc.info/gems/activesupport/3.2.22/ActiveSupport/Inflector#demodulize-instance_method) method to make it cleaner. But its usefulness is questionable, since it does almost the same thing (that is, drops everything up until after last `::` in given String object). :)

File: README.md [hunk 1]
```diff
@@ -32,6 +32,24 @@ get "/" do
 end
 ```
 
+### Use a custom serializer
+
+```ruby
+get "/" do
+  user = User.find("123")
+  render user, serializer: 'CustomUserSerializer'
```
Comment (spijet) [inline README.md]: This is exactly what I wanted to propose later today! :D Thank you so much!!

**Comment by spijet** [comment]
I have converted my app to use Grape and `grape_fast_jsonapi` for its API and it works perfectly in production environment. Thank you so much! :)

**Pull request merged**